src/Security/Authentication/FrontendPreviewAuthenticator.php

```diff
@@ -76,11 +76,12 @@ public function authenticateFrontendUser($username = null): void
         $request = $this->requestStack->getCurrentRequest();
 
         // Check if a back end user is authenticated
-        if (null === $this->tokenStorage->getToken() || !$this->tokenStorage->getToken()->isAuthenticated()) {
-            return;
-        }
-
-        if (null === $username || !$request->hasSession()) {
+        if (null === $username
+            || null === $request
+            || !$request->hasSession()
+            || null === $this->tokenStorage->getToken()
+            || !$this->tokenStorage->getToken()->isAuthenticated()
+        ) {
```

Comment:
multicondition `if`s are shitty to test.

Reply:
Oh yes. Just fire up the PhpStorm EA Extended Plugin and it will tell you. 😄 

I am going to address the null pointer issue in the next couple of days.

Reply:
🤔 good point, never thought of that. Should that be a general rule @contao/developers ?

Reply:
No idea what rule you're talking about so ¯\_(ツ)_/¯

Reply:
A general rule to not have too many conditions in an if statement, like splitting between "condition types", in this case token check and "others". I see @bytehead's point in unit test coverage which only works for separate lines.

Reply:
And it's not only about the tests. It's also bad to read and there's probably no enormous win of performance as well.